### PR TITLE
Extend client to get tasks

### PR DESCRIPTION
**Pull request by bpathak-ons**
### What

* Client for getTasks to get number to tasks assigned to a job

### How to review

* code sense check
* CI pass

### Who can review

* Anyone

### CODE REVIEW

File: sdk/v1/client_test.go [hunk 1]
```diff
@@ -679,3 +699,127 @@ func TestClient_GetTask(t *testing.T) {
...
+				So(tasks.TotalCount, ShouldEqual, 20)
+				So(tasks.TaskList[0].JobID, ShouldEqual, "883c81fd-726d-4ea3-9db8-7e7c781a01cc")
+				So(tasks.TaskList[0].TaskName, ShouldEqual, "zebedee")
+				So(tasks.TaskList[0].NumberOfDocuments, ShouldEqual, 10)
+				So(tasks.TaskList[1].JobID, ShouldEqual, "883c81fd-726d-4ea3-9db8-7e7c781a01cc")
+				So(tasks.TaskList[1].TaskName, ShouldEqual, "dataset-api")
+				So(tasks.TaskList[1].NumberOfDocuments, ShouldEqual, 20)
+			})
+
+			Convey("And an ETag is returned", func() {
```
Comment (cookel2) [inline sdk/v1/client_test.go:749]: Although an ETag is being returned by the test there is no ETag currently being returned, in the response header, if the GET tasks endpoint is actually called. So I think this part of the test should be removed as it is not currently doing this.

File: sdk/v1/client_test.go [hunk 2]
```diff
@@ -679,3 +699,127 @@ func TestClient_GetTask(t *testing.T) {
...
+			So(task, ShouldBeNil)
+
+			So(err.Error(), ShouldEqual, "failed as unexpected code from search reindex api: 500")
+			So(apiError.ErrorStatus(err), ShouldEqual, http.StatusInternalServerError)
+
+			Convey("Then the expected empty task is returned", func() {
+				So(task, ShouldBeNil)
+			})
+
+			Convey("And an empty ETag is returned", func() {
```
Comment (cookel2) [inline sdk/v1/client_test.go:782]: Similarly I think this part should be removed as not even an empty ETag is currently being returned.

File: sdk/v1/client_test.go [hunk 3]
```diff
@@ -679,3 +699,127 @@ func TestClient_GetTask(t *testing.T) {
...
+			So(err, ShouldNotBeNil)
+			So(task, ShouldBeNil)
+			So(err.Error(), ShouldEqual, "failed as unexpected code from search reindex api: 404")
+			So(apiError.ErrorStatus(err), ShouldEqual, http.StatusNotFound)
+
+			Convey("Then the expected empty task is returned", func() {
+				So(task, ShouldBeNil)
+			})
+
+			Convey("Then an empty ETag is returned", func() {
```
Comment (cookel2) [inline sdk/v1/client_test.go:811]: Same here.

**Review by cookel2 — CHANGES_REQUESTED** [review]
Just a few changes requested please.

File: sdk/v1/client_test.go (hunk 1 above)
Comment (bpathak-ons) [reply sdk/v1/client_test.go:749]: correct, trello task for generating ETag in response header is - https://trello.com/c/flHAokOt/1152-search-reindex-api-ifmatch-and-etag-headers-for-get-jobs-jobid-tasks-endpoint

Comment (nshumoogum) [reply sdk/v1/client_test.go:749]: I would right the client and unit test based on the swagger spec (assuming this is up to date)

In this case as Bhupendra says we will want to check the ETag is in the response header and not returned in the body. the data is mocked and so we can remove the ETag entry in the expectedTask variable when all the client methods have been updated.

See my next comment on removing duplicate assertions based on the above

File: sdk/v1/client_test.go [hunk 4]
```diff
@@ -679,3 +699,127 @@ func TestClient_GetTask(t *testing.T) {
...
+				So(tasks.TaskList[0].JobID, ShouldEqual, "883c81fd-726d-4ea3-9db8-7e7c781a01cc")
+				So(tasks.TaskList[0].TaskName, ShouldEqual, "zebedee")
+				So(tasks.TaskList[0].NumberOfDocuments, ShouldEqual, 10)
+				So(tasks.TaskList[1].JobID, ShouldEqual, "883c81fd-726d-4ea3-9db8-7e7c781a01cc")
+				So(tasks.TaskList[1].TaskName, ShouldEqual, "dataset-api")
+				So(tasks.TaskList[1].NumberOfDocuments, ShouldEqual, 20)
+			})
+
+			Convey("And an ETag is returned", func() {
+				So(respHeaders, ShouldNotBeNil)
```
Comment (nshumoogum) [inline sdk/v1/client_test.go:750]: These two assertions (line 750 and 751) can be removed as they are duplicating the same checks on ETag entry

```suggestion
```

File: sdk/v1/client_test.go [hunk 5]
```diff
@@ -679,3 +699,127 @@ func TestClient_GetTask(t *testing.T) {
...
+			So(task, ShouldBeNil)
+
+			So(err.Error(), ShouldEqual, "failed as unexpected code from search reindex api: 500")
+			So(apiError.ErrorStatus(err), ShouldEqual, http.StatusInternalServerError)
+
+			Convey("Then the expected empty task is returned", func() {
+				So(task, ShouldBeNil)
+			})
+
+			Convey("And an empty ETag is returned", func() {
```
Comment (nshumoogum) [inline sdk/v1/client_test.go:782]: Probably want to update the message in Convey:
```suggestion
			Convey("And no headers are returned", func() {
```

File: sdk/v1/client_test.go [hunk 6]
```diff
@@ -679,3 +699,127 @@ func TestClient_GetTask(t *testing.T) {
...
+			So(err, ShouldNotBeNil)
+			So(task, ShouldBeNil)
+			So(err.Error(), ShouldEqual, "failed as unexpected code from search reindex api: 404")
+			So(apiError.ErrorStatus(err), ShouldEqual, http.StatusNotFound)
+
+			Convey("Then the expected empty task is returned", func() {
+				So(task, ShouldBeNil)
+			})
+
+			Convey("Then an empty ETag is returned", func() {
```
Comment (nshumoogum) [inline sdk/v1/client_test.go:811]: Again probably want to update message in Convey func:

```suggestion
			Convey("Then no headers are returned", func() {
```

File: sdk/v1/client_test.go (hunk 1 above)
Comment (bpathak-ons) [reply sdk/v1/client_test.go:749]: ETag entry removed with duplicate assertion.

**Review by nshumoogum — APPROVED** [review]

**Review by cookel2 — APPROVED** [review]
looks good to me

**Pull request merged**